src/MissionManager/TransectStyleComplexItemTest.cc

```diff
@@ -226,21 +209,57 @@ void TransectStyleComplexItemTest::_testAltMode(void)
     QVERIFY(!_transectStyleItem->followTerrain());
 }
 
-TransectStyleItem::TransectStyleItem(PlanMasterController* masterController, QObject* parent)
+void TransectStyleComplexItemTest::_testFollowTerrain(void) {
+    _multiSpy->clearAllSignals();
+    _transectStyleItem->cameraCalc()->distanceToSurface()->setRawValue(50);
+    _transectStyleItem->setFollowTerrain(true);
+    _multiSpy->clearAllSignals();
+    while(_transectStyleItem->readyForSaveState() != TransectStyleComplexItem::ReadyForSave) {
+        QVERIFY(_multiSpy->waitForSignalByIndex(lastSequenceNumberChangedIndex, 50));
+    }
+    QList<double> expectedTerrainValues{497,509,512,512};
+    QCOMPARE(_transectStyleItem->transects().size(), 1);
+    for (const auto& transect : _transectStyleItem->transects()) {
+        QCOMPARE(transect.size(), 4);
+        for (const auto pt : transect) {
+            QCOMPARE(pt.coord.altitude(), expectedTerrainValues.front());
+            expectedTerrainValues.pop_front();
+        }
+    }
+}
+
+TestTransectStyleItem::TestTransectStyleItem(PlanMasterController* masterController, QObject* parent)
     : TransectStyleComplexItem      (masterController, false /* flyView */, QStringLiteral("UnitTestTransect"), parent)
     , rebuildTransectsPhase1Called  (false)
     , recalcComplexDistanceCalled   (false)
     , recalcCameraShotsCalled       (false)
 {
-
+    // We use a 100m by 100m square test polygon
+    const double edgeDistance = 100;
+    surveyAreaPolygon()->appendVertex(UnitTestTerrainQuery::linearSlopeRegion.center());
```

Comment:
this polygon lies in the slope region and thus in can count on emulated terrain values in the test cycle.